pkg/proto/configuration/blobstore/blobstore.proto

```diff
@@ -946,3 +953,14 @@ message DeadlineEnforcingBlobAccess {
   // The backend to which all operations are delegated.
   BlobAccessConfiguration backend = 2;
 }
+
+message CompressedClientConfiguration {
+  // Base GRPC client configuration. Must be a GRPC client that
+  // announces support for Zstd CAS compression.
+  buildbarn.configuration.grpc.ClientConfiguration grpc = 1;
+
+  // Minimum blob size in bytes to trigger compression.
+  // Blobs smaller than this threshold will use the non-compressed client
+  // for better performance on small objects.
+  int64 compression_threshold_bytes = 2;
```

Comment:
Is this feature necessary? Did you perform any measurements that justify adding this?

Reply:
This matches the configuration in Bazel, but yes, really small files are not worth the overhead to compress. Bazel defaults this to 100. We can either keep it at 100 bytes, or make it configurable.

For certain situations, it might be preferable to set it at something like 64k since the number of packets wont change much.

Here's the analysis: https://github.com/buildbuddy-io/buildbuddy/pull/7009#discussion_r1676117359